aiida/tools/graph/age_rules.py

```diff
@@ -0,0 +1,406 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Rules for the AiiDA Graph Explorer utility"""
+
+from abc import ABCMeta, abstractmethod
+from collections import defaultdict
+import numpy as np
+
+from aiida.common.exceptions import InputValidationError
+from aiida.orm.querybuilder import QueryBuilder
+from aiida.tools.graph.age_entities import Basket
+
+
+def check_if_basket(entity_set):
```

Comment:
this check can be replaced by the standard utility in `aiida.common.lang.type_check`. You can define the exception message you want if you don't like the default